File: Algo/hr_strings/reduceString.js
function reduce(s) {
  let obj = {};
  for (let i = 0; i < s.length; i++) {
    if (s[i] == s[i + 1]) {
      s = s.replace(s[i] + s[i + 1], '');
      obj = { all: false, val: s };
      return obj;
      break;
    }
  }
  obj = { all: true, val: s };
  return obj;
}

function superReducedString(s) {
  let allDone = false;
  while (allDone == false) {
    let obj = reduce(s);
    allDone = obj.all;
    s = obj.val;
  }

  if (s == '') {
    return 'Empty String';
  }

  return s;
}

console.log(superReducedString('aaabccddd'));

File: Algo/hr_strings/reduceString.cpp
#include <iostream>
#include <bits/stdc++.h>
using namespace std;

string superReducedString(string s)
{
    for (int i = 0; i < s.size(); i++)
    {
        if (s[i] == s[i + 1])
        {
            s.erase(i, 2);
            i = -1;
        }
    }

    if (s == "")
    {
        return "Empty String";
    }

    return s;
}

int main()
{
    //baab, aaabccddd
    cout << "Ans: " << superReducedString("baab") << endl;
    return 0;
}
